jonfroehlich/makeabilitylabwebsite
# Auto rename files (Talks and Publications pdfs)

Addresses Issue #257 
Before, the pdf_files of either the Talks or Publication objects will save the pdf_file locally with the original name of the file. 

Now, the file will be saved in the format `SpeakerLastName_Title_Year.pdf`. If there is more than one person attached to the instance object, the code will take the last name of the first person in the queryset. If there is whitespace in the title, it will be replaced with underscores. There are two functions that are set up to be triggered by Django's built-in `m2m_changed` signal. It will check that the object is being created and not updated, if so, it will change the file path and name of the pdf to the new format. 

To test:
- Run the server and add a talk object via the admin
- Save it and navigate to the media/talks directory locally to verify that the pdf name has been correctly changed
- Do the same publications
